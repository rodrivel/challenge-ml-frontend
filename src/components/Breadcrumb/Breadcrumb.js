import React from 'react';
import PropTypes from 'prop-types';
import cx from './Breadcrumb.module.scss';
import { ReactComponent as ArrowRight } from '../../assets/arrowRight.svg';

const breadcrumb = (props) => {
  const { categories } = props;  

  return (
    <ul className={cx.Breadcrumb}>
      { categories && categories.map((category, index) => <li className={cx.Breadcrumb__Item} key={`bc-${index}`}>{ index > 0 ? <ArrowRight/> : '' }<span>{category}</span></li> ) }      
    </ul>
  );
}

breadcrumb.propTypes = {
  categories: PropTypes.array.isRequired
}

export default breadcrumb;