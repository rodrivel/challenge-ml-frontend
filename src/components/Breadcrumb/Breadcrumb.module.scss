@import 'src/styles/base';

.Breadcrumb {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  margin: 16px 0;
  overflow-x: auto;

  .Breadcrumb__Item {     
    display: flex;
    align-items: center;
    color: $dark-gray;
    font-size: 14px;

    svg {
      fill: $dark-gray;
    }    

    span {
      white-space: nowrap;
    }

    &:last-child {
      span {
        font-weight: 500;
      }
    }
  }


}