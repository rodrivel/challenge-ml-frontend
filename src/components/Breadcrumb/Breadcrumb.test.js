import React from 'react';
import { shallow } from 'enzyme';
import Breadcrumb from './Breadcrumb';

it('expect to render Breadcrumb component', () => {
  const mockCategories = [
    'Celulares y Tel√©fonos',
    'Smartwatches y Accesorios',
    'Smartwatch',
  ];
  expect(shallow(<Breadcrumb categories={mockCategories} />)).toMatchSnapshot();
});
