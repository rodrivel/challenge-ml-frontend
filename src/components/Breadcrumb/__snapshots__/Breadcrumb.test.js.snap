// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Breadcrumb component 1`] = `ShallowWrapper {}`;
